### PR TITLE
Improvements to iteminfo constructor

### PR DESCRIPTION
**Pull request by jbytheway**
#### Summary
SUMMARY: Infrastructure "Better clarity in item description code"

#### Purpose of change
Currently many `iteminfo` objects are constructed to form an item description.  The constructor for `iteminfo` has four boolean and four string arguments, in a more or less arbitrary order.  It's very hard to understand what's going on from a call site.

To make the code clearer and facilitate future changes, I want to improve this situation.

#### Describe the solution
Changes to the constructor:
* Replace all four `bool` arguments with a single flag enum.
* Remove the `Plus` argument and add a new flag to replace most uses of it.
* Added an alternative overload for simple uses.

The `Plus` argument was used to prefix numbers with a `$` or `+`.  However, these were both bad in different ways which these changes should help to resolve.
* Adding `$` as a separate string causes translation issues.  This PR fixes the first point of #23481.
* Adding `+` means that it is formatted differently from `-` (which is printed as part of the number).  In the new code, we print `+` in the same way as `-`, via the standard string formatting options, yielding a consistent appearance.

#### Additional context
I also fixed the formatting of gun sight dispersion, which was being done differently from all the other gun stats.

### CODE REVIEW

File: src/output.cpp [hunk 1]
```diff
@@ -862,13 +861,23 @@ std::string format_item_info( const std::vector<iteminfo> &vItemDisplay,
                         }
                     }
                 }
-                buffer << sPlus << "<color_" << string_from_color( thisColor ) << ">";
-                if( vItemDisplay[i].is_int ) {
-                    buffer << string_format( "%.0f", vItemDisplay[i].dValue );
+                const char *fmt;
```
Comment (BevapDin) [inline src/output.cpp]: Please use `std::string`,the code is written as C++ and not C.

File: src/item.h [hunk 2]
```diff
@@ -133,23 +130,42 @@ struct iteminfo {
...
+        };
+
         /**
          *  @param Type The item type of the item this iteminfo belongs to.
          *  @param Name The name of the property this iteminfo describes.
+         *  @param flags Additional flags to customize this entry
          *  @param Fmt Formatting text desired between item name and value
          *  @param Value Numerical value of this property, -999 for none.
-         *  @param _is_int If true then Value is interpreted as an integer
          *  @param Plus Character to place before value, generally '+' or '$'
```
Comment (BevapDin) [inline src/item.h]: This parameter is also gone (or was never there).

Comment (jbytheway) [reply src/item.h]: Thanks, fixed.

File: src/output.cpp (hunk 1 above)
Comment (jbytheway) [reply src/output.cpp]: I'm not sure I agree, but it doesn't matter; this code is unnecessary anyway and I've now removed it.

**Comment by jbytheway** [comment]
Rebased to resolve merge conflicts.

File: src/item.cpp [hunk 3]
```diff
@@ -1218,24 +1244,28 @@ std::string item::info( std::vector<iteminfo> &info, const iteminfo_query *parts
         damage_instance ammo_dam = has_ammo ? curammo->ammo->damage : damage_instance();
         // @todo This doesn't cover multiple damage types
         int ammo_pierce     = has_ammo ? get_ranged_pierce( *curammo->ammo ) : 0;
-        int ammo_range      = has_ammo ? curammo->ammo->range      : 0;
+        //int ammo_range      = has_ammo ? curammo->ammo->range      : 0;
```
Comment (kevingranade) [inline src/item.cpp]:
```suggestion
```
Just delete instead of commenting out.

Comment (jbytheway) [reply src/item.cpp]: Ah, I forgot about that.  I had intended to go back and see whether a new item should be added which listed the ammo range somehow, but that's probably a problem for another day.  I'm happy to delete it.

**Pull request merged**